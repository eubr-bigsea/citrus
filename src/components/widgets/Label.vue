<template>
    <span class="label">
        <span :data-order="field.order" :data-name="field.name"
            :class="{'text-danger': error, required: field.required}" :title="error">{{field.label || field.name}}</span>
        <span v-show="field.required">*</span>:
        <span v-if="(showHelp === undefined || showHelp) && field.help" class="data-help">
            <span class="fa fa-question-circle float-right" :title="field.help"></span>
        </span>
    </span>
</template>

<script>
    import Widget from '../../mixins/Widget.js';
    export default {
        mixins: [Widget],
        name: "property-label",
        props: {
            field: { required: true, type: Object },
            value: {},
            showHelp: true,
        },
        computed: {
            error() {
                if (this.field.required && (this.value === undefined || this.value === null || this.value === '' || this.value === {})) {
                    return this.$tc('errors.missingRequiredValue');
                }
                return null;
            },
        }
    }
</script>
<style scoped>
    .label {
        font-size: 1em;
    }
</style>
