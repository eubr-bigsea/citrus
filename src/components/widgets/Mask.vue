<template>
    <div>
        <LabelComponent :field="field" />
        <div v-if="readOnly">
            {{value === null ? field.default: value}}
        </div>
        <div v-else>
            <input type="text" v-mask="mask" class="form-control" />
        </div>
    </div>
</template>
<script>
    import LabelComponent from './Label.vue'
    import Widget from '../../mixins/Widget.js';
    import { debounce } from '../../util.js';
    export default {
        mixins: [Widget],
        components: { LabelComponent },
        props: {
            mask: { type: Array, default: () => [] }
        },
        data() {
            return {
                internalValue: ''
            }
        },
        mounted(){
            // this.internalValue = this.value;
        },
        methods: {
            updated: debounce(function (e) {
                this.$root.$emit(this.message, this.field, e.target.value);
            }, 500)
        },
    }
</script>