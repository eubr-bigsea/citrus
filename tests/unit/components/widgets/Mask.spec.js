import { shallowMount } from '@vue/test-utils';
import { expect, it } from 'vitest';
import Label from '$SRC/components/widgets/Label.vue';
import Mask from '$SRC/components/widgets/Mask.vue';

describe('Mask.vue', () => {
  it('ReadOnly True', () => {
    const wrapper = shallowMount(Mask, {
        propsData: {
            field: {
                value: null,
                default: 'User input'
            },
            readOnly: true,
            value: null,
        },
    });

    expect(wrapper.findComponent(Label).exists()).toBe(true);
    expect(wrapper.text()).toBe("User input");
  });

  it('ReadOnly false', () => {
    const wrapper = shallowMount(Mask, {
        propsData: {
            field: {
                value: 'Text', default: 0
            },
            readOnly: false, value: null
        }
    });

    expect(wrapper.findComponent(Label).exists()).toBe(false);
    const masktext = wrapper.find('masktext');
    expect(masktext.exists()).toBeTruthy();
  });
  
  it('snapshots the component', () => {
    const wrapper = shallowMount(Mask, { propsData: { field: {} } });
    expect(wrapper.text()).toMatchSnapshot();
});
});
