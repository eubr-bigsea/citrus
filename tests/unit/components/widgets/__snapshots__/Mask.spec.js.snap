// Vitest Snapshot v1

exports[`Mask.vue > snapshots the component 1`] = `""`;
